tests/src/test/scala/actionContainers/PythonActionContainerTests.scala

```diff
@@ -140,6 +141,76 @@ class PythonActionContainerTests extends BasicActionRunnerTests with WskActorSys
         })
     }
 
+    it should "run zipped Python action containing a virtual environment" in {
+        val zippedPythonActionWrongName = TestUtils.getTestActionFilename("action_python_virtualenv.zip")
+
+        val code = readAsBase64(Paths.get(zippedPythonActionWrongName))
+
+        val (out, err) = withActionContainer() { c =>
+            val (initCode, initRes) = c.init(initPayload(code, main = "main"))
+            initCode should be(200)
+            val args = JsObject("msg" -> JsString("any"))
+            val (runCode, runRes) = c.run(runPayload(args))
+            runCode should be(200)
+            runRes.get.toString() should include("netmask")
+        }
+        checkStreams(out, err, {
+            case (o, e) =>
+                o should include("netmask")
+                e shouldBe empty
+        })
+    }
+
+    it should "report error if zipped Python action has wrong main module name" in {
+        val zippedPythonActionWrongName = TestUtils.getTestActionFilename("action_python_virtualenv_name.zip")
```

Comment:
could you have reused the same zip file as above but passed `main = "foobar"` below?